### ISSUE TITLE
7.1.3.2 may have broken stimulus controllers

### ISSUE THREAD

**Issue by salex** [rails/rails#51160]
### Steps to reproduce

I happen to be creating a new Rails test app a few hours ago  when 7.1.3.2 was released. 

A stimulus controller I've used for years all of a sudden produced an error

```ruby
The resource http://localhost:3000/assets/controllers/flashIt.controller-
d7fc6bd02d113b8faec8c95d343483bb35099e6a6f269f9a13f0de5796578850.js 
was preloaded using link preload but not used within a few seconds from the window's load event.
 Please make sure it wasn't preloaded for nothing.
```

The markup that loads the controller

```slim
div.col-span-6
      - flash.each do |name, msg|
        - if msg.is_a?(String)
          .flash-box.w3-animate-right.p-2[
            data-controller="flashIt" 
            data-flashIt-target="flashBox" 
            class="#{name.to_s == 'notice' ? 'text-green-500 bg-green-100 px-4' : 'p-2 text-red-500 bg-red-100 px-4'}"
          ]
            = content_tag :span, content_tag(:strong,msg)
```

The controller

```javascript
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {
  static targets = [ "flashBox" ]

  connect() {
    this.closeFlashIt()
  }

  closeFlashIt(){
    var box = this.flashBoxTarget
    setTimeout(function () {
      var op = 1;  // initial opacity
        var timer = setInterval(function () {
            if (op <= 0.1){
                clearInterval(timer);
                box.style.display = 'none';
            }
            box.style.opacity = op;
            box.style.filter = 'alpha(opacity=' + op * 100 + ")";
            op -= op * 0.1;
        }, 500);
    }, 10000);
  }
}
```
The flash message will show up if there is a flash message, but it won't fade out.

The message first shows up in the js console shortly after the page is loaded (it's on every page since it's in the menubar. I opened the console when the message did not fade out.

### System configuration
**Rails version**: 7.1.3.2

**Ruby version**: 3.2

**Comment by salex** [comment]
I'll just add that the console.log message does not show up. It's not being called.

**Comment by rafaelfranca** [comment]
Nothing in that release might have broken how assets are loaded. Did you try to downgrade to check if it is really 7.1.3.2?

**Comment by salex** [comment]
The cuprit is me, I'll explain in a minute.

I created a new rails app using all the defaults + database and css tailwind.

* I created a home controller with only index and show actions
* Created a menubar with link buttons to the home and show actions and my flash div
* Added the stimulus flashIt controller
* Added  **flash.now.notice = "You are now showing something"** to the show action

Clicking on the show button worked. (showed the flash message in the menubar). No load error.

The explanation is that I was toying with the idea or rewriting a 10 year old app from scratch. While I kept it up to date, it still had a lot of old stuff. Convoluted methods, poorly written  CSS, etc.

The new app I created (actually several times!) skipped some stuff (jbuilder, action_text, etc) that I didn't think I needed.

Apparently I skipped something that didn't mix well with Stimulus. Before the Stimulus problem I had other problems I could not explain. It took a while to notice that if running bin/dev and I hovered over a link, I could see a query being executed in the console. I first thought it may of been something new, but it was actually doing queries for the link I was hovering over. Even saw errors in query that were not display.

That was the reason for trying new apps several times.

I have no earthly idea on what I skipped that would cause the problems I was seeing.  Things like something redirecting to the root path, TURBO would render and add to the index versus replacing it.

I guess I won't try skipping stuff that I'm not sure where its linked into the Rails.

Sorry for the bad issue, I'll close it.
